### ISSUE TITLE
componentWillReceiveProps not recomended to use.

### ISSUE THREAD

**Issue by bartekcis** [mukeshsoni/react-telephone-input#283]
Hi,

It seems that the library uses life cycle methods no longer considered ass a good practice. When I run Jest test I get warnings:

_console.warn node_modules/react-dom/cjs/react-dom.development.js:88
      Warning: componentWillReceiveProps has been renamed, and is not recommended for use. See https://fb.me/react-unsafe-component-lifecycles for details.

      * Move data fetching code or side effects to componentDidUpdate.
      * If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://fb.me/react-derived-state
      * Rename componentWillReceiveProps to UNSAFE_componentWillReceiveProps to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.

      Please update the following components: ReactTelephoneInput_

Those are my core modules:
"react": "^16.12.0",
    "react-app-polyfill": "^1.0.6",
    "react-dom": "^16.12.0",
    "react-router-dom": "^5.1.2",
    "react-scripts": "^3.4.0"
    "react-telephone-input": "^4.74.4",

Do you think the module can be updated so those warnings are no longer visible?

Thanks,
Bart

**Comment by mukeshsoni** [comment]
Which version of react-telephone-input are you using?

**Comment by bartekcis** [comment]
It is "react-telephone-input": "^4.74.4",

**Comment by mukeshsoni** [comment]
Can you try 4.75.4?

**Comment by bartekcis** [comment]
Still no good but it's a step forward I think. Now it complains about `VirtualList` component:
![image](https://user-images.githubusercontent.com/57143717/77315735-f7ea2900-6cff-11ea-9da0-c13a1cba7dfe.png)

**Comment by mukeshsoni** [comment]
Ok. I am using `react-tiny-virtual-list` to render the list internally. That component uses `componentWillReceiveProps`. I can't change that component at this point in time.

**Comment by bartekcis** [comment]
Ok. Thanks, do you think it's worth to rise an issue there? Are you going to release 4.75.4?

**Comment by mukeshsoni** [comment]
I have released 4.75.4. Didn't you install 4.75.4 through npm?

I would suggest raising an issue with react-tiny-virtual-list. If possible, raise a PR with an alternative.

**Comment by bartekcis** [comment]
Yes, I did it via npm. Ok I'll rise an issue there. Thanks for help!
